ui/src/shared/components/tables/TableGraph.tsx

```diff
@@ -71,7 +71,9 @@ class TableGraph extends PureComponent<Props, State> {
       sortOptions.field = fieldName
       sortOptions.direction = DEFAULT_SORT_DIRECTION
     }
-    this.setState({sortOptions})
+
+    const newSortOptions = {...sortOptions}
+    this.setState({sortOptions: newSortOptions})
```

Comment:
Why not use the setState updater function here and put everything into the callback, which will update immediately? For reference see: https://reactjs.org/docs/react-component.html#setstate Something like this, which may need to be cleaned up with prettier:

    public handleSetSort = (fieldName: string) => {
      this.setState(({sortOptions}) => {
        const newSortOptions = {...sortOptions}
        if (fieldName === sortOptions.field) {
          newSortOptions.direction =
            sortOptions.direction === ASCENDING ? DESCENDING : ASCENDING
        } else {
          newSortOptions.field = fieldName
          newSortOptions.direction = DEFAULT_SORT_DIRECTION
        }
        return {newSortOptions}
      })
    }

Reply:
I'm not following. I'm cheating by having offline context, but this change is to fix a condition in which data within sortOptions had changed, but the reference was never updated for `sortOptions` so react didn't know about it. Cool reference though! Maybe i'm still wrapping my head around it.

Reply:
setState takes an object or a function. Using a function makes the component update immediately, instead of possibly waiting (when using an object) due to React batching the state updates.

When a user updates the sort options, they expect to see something update immediately, which seems like a good thing to guarantee with the setState updater function.

Reply:
👍 thanks for helping me learn

Reply:
I tried to do this in the code, and it was complaining about sort options not being assignable, so for now ill just keep what we have :)

Reply:
i edited my snippet. I left off the correct key in the updated state